### ISSUE TITLE
eCommerce: My site drawer shows the wrong plan info

### ISSUE THREAD

**Issue by alaczek** [Automattic/wp-calypso#38499]
When viewing the homepage of a site on eCommerce plan, the My Site button in the top left corner shows what looks like outdated version of the drawer, but I also noticed it says the site is on Free plan, when in fact it is on eCommerce plan (as reflected in Calypso).

#### Steps to reproduce
1. Create a new site on an eCommerce plan.
2. View the homepage of your new site.
3. Click on My site in the top left corner - it says "Free" next to the plan.
![image](https://user-images.githubusercontent.com/4550351/71049207-d116d680-2195-11ea-936b-b528a093e162.png)

4. Click on any of the links in the drawer to go to Calypso. It says "eCommerce" next to Plan.
![image](https://user-images.githubusercontent.com/4550351/71049227-dc6a0200-2195-11ea-8237-ad98e7e0a934.png)

#### What I expected
For the correct plan info to be displayed in both places.

#### Context / Source
#dogfooding, #manual-testing

**Comment by timmyc** [comment]
Confirmed this bug is still present.

I think the version of the masterbar in use here is from Jetpack, so I imagine [this logic](https://github.com/Automattic/jetpack/blob/master/modules/masterbar/masterbar.php#L907-L917) might be where to investigate this issue further.

**Comment by lancewillett** [comment]
Interesting find — the "hover" menu in masterbar is deprecated for Simple sites. Links now go directly to Calypso instead of showing the dropdowns.

For history's sake, it became too much work to maintain the separate menus, and removing them is the simplest solution.

**Comment by timmyc** [comment]
> Interesting find — the "hover" menu in masterbar is deprecated for Simple sites.

@lancewillett so the hover menu should be removed from the JP masterbar then?

**Comment by lancewillett** [comment]
Yes, it should be removed everywhere now. Thanks for catching this.

**Comment by davipontesblog** [comment]
I suggest we remove this nav drawer from atomic sites as well, and simply redirect users to "My Home" when clicking on "My Site". Much better experience.

**Comment by lancewillett** [comment]
@davipontesblog +100 internal ref: pbAPfg-3G-p2

**Comment by timmyc** [comment]
> I suggest we remove this nav drawer from atomic sites as well, and simply redirect users to "My Home" when clicking on "My Site". Much better experience.

I _think_ the nav drawer on the AT sites *is* the Jetpack version - at least that is what my explorations led me to.

**Comment by gwwar** [comment]
@timmyc were you working on this?

**Comment by timmyc** [comment]
@gwwar - no I am not. I assign myself when I'm working on an issue - sorry if I gave the signal that I was working on this. Depending on how things go during our current cool down period, I'm hoping to carve some time to work on more ecomm issues logged here though.

**Comment by obenland** [comment]
I've put https://github.com/Automattic/wp-calypso/issues/50519 on the project board for Manage, which will remove the My Site drawer. We'll close this ticket once that's done. /cc @simison

**Comment by m1r0** [comment]
It looks like this has been solved. The my site drawer is no longer there. Closing...
